src/serena/dashboard.py

```
@@ -317,6 +318,36 @@ def get_last_execution() -> dict[str, Any]:
+            except Exception as e:
+                return {"status": "error", "message": str(e)}
+
+        @self._app.route("/mark_news_snippet_as_read", methods=["POST"])
+        def mark_news_snippet_as_read() -> dict[str, str]:
+            try:
+                request_data = request.get_json()
+                news_snippet_id = int(request_data.get("news_snippet_id"))
+                news_snippet_id_file = SerenaPaths().news_snippet_id_file
+                with open(news_snippet_id_file, "w", encoding="utf-8") as f:
+                    f.write(str(news_snippet_id))
+                return {"status": "success", "message": f"Marked news snippet {news_snippet_id} as read"}
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)